proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1028 @@
+        resource:
+          "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+          … baz's ClusterLoadAssignment payload …
+```
+
+
+Note that the first resource bar can be referenced in a URI as
+`xdstp://some-authority/envoy.config.listeners.v3.ListenerCollection/foo#entry=bar`,
+while the second resource is anonymous and cannot be referenced outside the
+collection.
+
+#### Glob collections
+
+Client bootstrap:
+
+
+```
+dynamic_resources:
+  lds_resources_locator:
+    name: xdstp://some-authority/envoy.config.listeners.v3.Listener/my-listeners/*?node_type=ingress
+  lds_config:
+    authorities:
+    - name: some-authority
+  … rest of ConfigSource pointing at xDS management server … 
+```
+
+
+Client LDS `DeltaDiscoveryRequest` sent to xDS relay proxy (note the use of client capabilities):
+
+
+```
+resource_names_subscribe:
+- xdstp://some-authority/envoy.config.listeners.v3.Listener/my-listeners/*?xds.node_type=ingress&xds.client_features.envoy.config.no_bind_to_port=true
+```
+
+xDS management server `DeltaDiscoveryResponse` sent to the client:
+
+
+```
+resources:
+- version: 1
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/my-listeners/foo?xds.node_type=ingress&xds.client_features.envoy.config.no_bind_to_port=true
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … foo's Listener payload … 
+- version: 42
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/my-listeners/bar?xds.node_type=ingress&xds.client_features.envoy.config.no_bind_to_port=true
+  resource:
+    "@type": type.googleapis.com/xds.core.v3.ResourceLocator
+    <xdstp://some-authority/envoy.config.listeners.v3.Listener/baz>
+```
+
+
+In this example,
+`xdstp://some-authority/envoy.config.listeners.v3.Listener/my-listeners/bar` was
+redirected to `xdstp://some-authority/envoy.config.listeners.v3.Listener/baz.` The
+context parameters (ingress, client caps) were used to filter down the listener
+set for the client in a given collection (`my-listeners`).
+
+## Alternatives
+
+In this example, a CPaaS wants to specify on-premise control plane resources as
+failovers on network partition.
+
+The CPaaS management server `DeltaDiscoveryResponse` sends to the client inside a `Cluster` resource:
+
+```
+name: some-cluster
+type: EDS
+eds_cluster_config:
+  service_name: xdstp://some-cloud-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/foo#alt=xdstp://some-onprem-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/bar
+  eds_config:
+    authorities:
+    - name: some-cloud-authority
+  … rest of ConfigSource pointing at xDS management server … 
+```
+
+The client has a `ConfigSource for `some-onprem-authority` specified in its
+bootstrap.
+
+Client EDS `DeltaDiscoveryRequest` sends to xDS management server for `some-cloud-authority`:
+
+
+```
+resource_names_subscribe:
+- xdstp://some-cloud-authority/envoy.config.endpoint.v3.ClusterLoadAssignment/foo
+```
+
```

Comment:
can you expand on this. What does this mean?

Reply:
Which bit is unclear? Happy to expand on anything here, just need to figure out what is missing :)

Reply:
How is the fallback to 'some-onprem-authority' configured?

Reply:
The presence of the `#alt` directive above describes the fallback relationship. It's also mentioned that there is a mapping from `some-onprem-authority` to `ConfigSource` in the bootstrap.